### ISSUE TITLE
Mass-flow facts on the wire: why the own cell's mass moves (#324)

### ISSUE THREAD

**Issue by preffect** [preffect/evolution#383]
**Epic:** #100 · decision #324 (C + Z1), spec #356 (PR #389)

**Spec files:**
- `docs/ui/hud.md` §3.1.5 (the "facts the cues read" block and "Applied, at the floor and the cap")
- `docs/ecology/mass-and-movement.md` §4–§4.1 (including the #154 swallowed dose)
- `docs/architecture/wire-contract.md` §4.1
- `docs/architecture/entity-model.md` §2

The legibility cues show why the own cell's mass moves, but the snapshot carries only the mass. The server must report what it applied, so the client never keeps a second copy of the rules. The shape is gameplay-qa's wire proposal (https://github.com/preffect/evolution/issues/383#issuecomment-5673389967), adopted in hud.md §3.1.5; the architect confirms it on this ticket and writes it into `wire-contract.md` §4.1 and `entity-model.md` §2.

**Do**
1. **`OwnProgress.massFlow: MassFlowView | null`**, fields:
   - `ratesPerSecond: Partial<Record<'toxin' | 'swallowed' | 'decay' | 'vent' | 'light', number>>`: post-floor, post-cap, losses negative, zero causes omitted, `SNAPSHOT_MASS_RATE_DECIMALS` = 2. At 0.1 place, decay rounds to 0 below mass 45.
   - `decayTraitShare?`: the product of the owned `decayMultiplier` values, minus 1.
   - `zone`: the zone metabolism used this tick.
   - `sprintSpent?`: the mass a sprint start took, as applied.
2. **Where it is written.** A transient per-player record beside `world.effects`, filled from the server's own deltas in `metaboliseCell` and the sprint start. The swallowed dose is applied inside metabolism under the joint floor (`ecology/mass-and-movement.md` §4.1), not by the engulf step; the pro-rata split at the floor depends on that. It is never a `CellRecord` field, never hashed and never replayed, so the state hash is unchanged.
3. **At the floor.** `loss = massAtStart − massAfterFloor` is split pro rata across toxin, swallowed, decay and vent. `light = massAfterGain − massAfterFloor`, taken after the cap.
4. **`swallowed`.** The #154 dose: `prey.mass × prey.toxinDrainFractionPerSecond × ENGULF_SWALLOWED_TOXIN_MULTIPLIER + prey.mass × prey.spikeDrainFractionPerSecond`. It replaces that prey's contact toxin, and is never reported under `toxin`. If #154's code has not landed, the cause is wired and reads absent.
5. **Effect amounts.**
   - `EatEffect` gains `massGained` and `dnaGained`; `CellAbsorbedEffect` gains `predatorMassGained` and `predatorDnaGained`.
   - They are measured as deltas around `gainMass` / `gainDna`, cap overflow in the DNA, at `SNAPSHOT_MASS_DECIMALS` (0.1).
6. **Moves to shared.** Move `zoneAt`, `isInsideGelPatch` and `zoneDecayMultiplier` into `shared/src/simulation/zones.ts`, and extract `sprintMassCost` = `min(mass × SPRINT_MASS_COST_FRACTION, mass − CELL_STARTING_MASS)` into shared. These are pure moves.
7. **Bytes.** Measure with grazer bots (the #339 method): `massFlow` is about 70 B to its own viewer; the eat amounts are about 32 B per eat to every viewer. Compare against the snapshot budget.

**Acceptance** (the #212 rule: every expectation computed from balance constants in the test, never a literal such as `9.36` or `−0.2`):
- **Unit.** Each rate against the metabolism formula at the worked setup (mass 312, in the vent, Mitochondrion I, touching Toxin Vacuole I). The vent share is the trait-scaled broth share × (`VENT_DECAY_MULTIPLIER` − 1).
- **Conservation gameplay scenario** (`*.gameplay.test.ts`, a new acceptance row in `ecology/acceptance.md`). On every tick, before rounding:

  `Δmass == Σ ratesPerSecond × TICK_INTERVAL_S + Σ own massGained + predatorMassGained − sprintSpent`

  Setups:
  - vent + Mitochondrion I
  - contact with Toxin Vacuole I
  - an aura toxin with no contact
  - Chloroplast in the shallows
  - the floor (mass 20.01 touching toxin)
  - the cap (5000 eating)
  - engulfing a toxic / spiny prey (the swallowed dose, once #154 lands)
- The state hash and replays are unchanged.

Unblocks #385 (the cues) and #387 (the Tab panel).

**Comment by preffect** [comment]
### Wire proposal for #383 (gameplay-qa)

Checked against `metabolism.ts`, `cell-mass.ts`, `engulf-payout.ts`, `eating.ts`, `inputs.ts` at d7740cf, plus the rulings on `origin/main` (#351 #179/#271, #375 #154).

**W1 · major · the cause list misses the swallowed dose (#154, on main since #375).** `ecology/mass-and-movement.md` §4.1 on main now has `swallowedDose = prey.mass × prey.toxinDrainFractionPerSecond × ENGULF_SWALLOWED_TOXIN_MULTIPLIER (8) + prey.mass × prey.spikeDrainFractionPerSecond`, and it replaces that prey's contact toxin. Neither part is in code yet (`ENGULF_SWALLOWED_TOXIN_MULTIPLIER` is still 6, and metabolism has no spike term). `MASS_RATE_CAUSE` has `toxin` and `spines` but never says whether the swallowed dose counts under them. Contact toxin says "back off" and a swallowed dose says "this meal is costing you", so they are different lessons (see C1 on #389). Recommend a separate `swallowed` cause (toxin dose plus spikes, or split into two), plus a "Done when" test for it.

**W2 · major · "applied" is undefined at the floor.** `loseMassToFloor` clips the combined decay + drain at `CELL_STARTING_MASS`. The shape does not say whether a rate is the formula's value or what was actually applied. A mass-20.3 cell touching Toxin Vacuole I would show `−0.6/s TOXIN` while losing 0.3 total. For "can't drift" to hold, the server has to report what it applied:
- `loss = massAtStart − massAfterFloor`, split across the requested losses in proportion.
- `light = massAfterGain − massAfterFloor` (after the cap).

**W3 · major · compute it from the server's deltas, not a second formula.** Yes, this is possible, and it keeps the state hash unchanged:
- `metaboliseCell` already has `decayPerSecond` and `drain` separately. It writes a transient per-player `massFlow` record next to `world.effects`, which is not hashed or replayed (#383 requires identical hashes, so it must not be a `CellRecord` field).
- The engulf step adds the swallowed dose once #154 lands.
- The effect amounts are measured the same way: cell mass after minus before around `gainMass`, and `dnaCumulative` after minus before around `gainDna`. For the payout that covers the yield, the cap overflow, the base and the earned-only share (#271) in one read, so no caller re-implements the modifiers.

**W4 · minor · rounding.** #357 quantises cell `mass` to 0.1 but does not touch `ownProgress`. If the rates were rounded to 0.1:
- broth decay `(m − 20) × 0.002` rounds to 0 below mass 45 and reads 0.1 up to about mass 95;
- the vent share (half of decay) rounds to 0 below about mass 70;
- the causes no longer add up to the chip's net rate.

Recommend `SNAPSHOT_MASS_RATE_DECIMALS` = 2, which costs about 3 B a cause. The display threshold stays `RATE_TAG_MIN_MASS_PER_SECOND` 0.1 on the client, so tiny rates are hidden by design, not by the wire. Effect amounts can use `SNAPSHOT_MASS_DECIMALS` (0.1).

**W5 · minor · the sprint floater is client-derived.** The spec computes `sprintMassCost(mass before)` from the rising edge of `sprintRemainingTicks`. That mass is up to `SNAPSHOT_EVERY_TICKS` = 3 ticks old, quantised, and may include that window's eats and decay, so it is not the server's number. Also, `sprintMassCost` must return the clipped amount `min(mass × fraction, mass − CELL_STARTING_MASS)`, so a spawn-mass cell shows no `−0 SPRINT`. Recommend the server report it: either `massFlow.sprintSpent` on the start tick, or a `sprint_started { cellId, massSpent }` effect (about 60 B, at most once per 3 s per cell). Moving `sprintMassCost` to shared is still fine.

**W6 · minor · zone.** Moving `zoneAt` / `isInsideGelPatch` / `zoneDecayMultiplier` to shared is a pure move and fine (the gel patches are already on `game_state`). But the pill and the zone beat read the predicted position, while the `VENT` rate reads the server's position, so on a boundary they can disagree for a few ticks. Carrying `zone: ZoneId` in `massFlow` (about 20 B) makes the pill, the beat and the VENT tag agree exactly.

**W7 · nit · the trait share.** `decayTraitShare` is the product of every owned trait's `decayMultiplier`, minus 1. With Mitochondrion I plus Chloroplast I it is 0.85 × 0.9 − 1 = −0.235. That total has no single "owning trait" for the glyph, so name the rule (for example, the largest contributor's glyph).

**Bytes.** Measured with `JSON.stringify` at the worked example:
- The full record is 123 B.
- Sparse (zero causes omitted, 2 dp) is about 70 B. That is sent to its own viewer only (`VIEWER_SNAPSHOT_KEYS`), about 1.4 KB/s at 20 Hz, 0.3 % of the 24 KB snapshot budget.
- `massGained` + `dnaGained` add about 32 B per `eat` effect, and effects go to every viewer. With 8 grazers in bloom (about 5–15 eats a window) that is about 0.2–0.5 KB a snapshot. That fits, but #383 should measure it with grazer bots (the #339 method), not estimate it.

**Recommended shape**
```ts
export const MASS_RATE_CAUSE = { toxin: 'toxin', swallowed: 'swallowed', decay: 'decay', vent: 'vent', light: 'light' } as const;
export interface MassFlowView {
  /** mass/s this tick's metabolism APPLIED (post-floor, post-cap), losses negative; a zero cause is omitted. 2 dp. */
  ratesPerSecond: Partial<Record<MassRateCause, number>>;
  /** product of owned decayMultipliers − 1; omitted when 0. */
  decayTraitShare?: number;
  /** the zone metabolism used this tick. */
  zone: ZoneId;
  /** mass the sprint start took this broadcast window; omitted when none. */
  sprintSpent?: number;
}
// EatEffect: massGained, dnaGained; CellAbsorbedEffect: predatorMassGained, predatorDnaGained — measured as deltas, 0.1 dp.
```
If `spines` stays its own cause, split `swallowed` into `swallowedToxin` / `spines`. The point is that contact toxin and the swallowed dose must be told apart.

**Acceptance for #383 (the #212 rule).**
- Every expected value comes from balance constants in the test, never the literal `9.36` or `−0.3` (the doc's worked vent value is already wrong, see M1 on #389).
- Add a **conservation scenario** (`*.gameplay.test.ts`, with a new acceptance row). On every tick, and before rounding: `Δmass == Σ rates × TICK_INTERVAL_S + Σ own eat massGained + predatorMassGained − sprintSpent`.
- Cover these setups: vent + Mitochondrion I, contact with Toxin Vacuole I, Chloroplast in the shallows, the floor (mass 20.01 touching toxin), the cap (5000 eating), and engulfing a toxic / spiny prey once #154 lands.
